src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1465,6 +1478,37 @@ protected void ExecuteIndexing()
             }
         }
 
+        private void FinishStaggeredDeploymentForThisNode(IndexDefinitionBase definition)
+        {
+            var nodeTag = DocumentDatabase.ServerStore.NodeTag;
+            var record = DocumentDatabase.ReadDatabaseRecord();
+
+
+            if (record.StaggeredIndexes.TryGetValue(definition.Name, out var staggeredIndex) == false)
+                return;
+
+            if (staggeredIndex.TryGetValue(nodeTag, out var staggeredIndexStatus) == false)
+                return;
+
+            if (staggeredIndexStatus != StaggeredIndexStatus.Running)
+                return;
+
+            // todo We need to make sure we only run this command once for each finished node
+            Task.Run(async () =>
+                {
+                    try
+                    {
+                        await DocumentDatabase.ServerStore.SendToLeaderAsync(new PutStaggeredIndexCommand(record.DatabaseName, Definition.Name, RaftIdGenerator.NewId()))
+                            .ConfigureAwait(false);
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(e);
```

Comment:
Should remove this.

Note that the current `throw` will _kill the server_ if there is an error here, which is expected.